pilot/pkg/model/validation.go

```diff
@@ -39,14 +39,18 @@ import (
 
 const (
 	dns1123LabelMaxLength int    = 63
-	dns1123LabelFmt       string = "[a-zA-Z0-9]([-a-z-A-Z0-9]*[a-zA-Z0-9])?"
+	dns1123LabelFmt       string = "[a-zA-Z0-9](?:[-a-z-A-Z0-9]*[a-zA-Z0-9])?"
 	// a wild-card prefix is an '*', a normal DNS1123 label with a leading '*' or '*-', or a normal DNS1123 label
 	wildcardPrefix = `(\*|(\*|\*-)?` + dns1123LabelFmt + `)`
 
 	// Using kubernetes requirement, a valid key must be a non-empty string consist
 	// of alphanumeric characters, '-', '_' or '.', and must start and end with an
 	// alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345'
-	qualifiedNameFmt string = "([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]"
+	qualifiedNameFmt string = "(?:[A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]"
+
+	// In Kubernetes, label names can start with a DNS name followed by a '/':
+	dnsNamePrefixFmt       string = dns1123LabelFmt + `(?:\.` + dns1123LabelFmt + `)*/`
```

Comment:
I am not great with regex, does this allow `foo.com/bar.com`? And if so, should it?

Reply:
yep! that’s valid as far as Kubernetes is concerned.

Reply:
Interesting. I wonder if we can just use their validation logic? Or point to it from here for reference possibly? Or are you seeing that just from trying it out

Reply:
we could use [`IsQualifiedName`](https://github.com/kubernetes/apimachinery/blob/master/pkg/util/validation/validation.go#L42) from the k8s `apimachinery` repo

Reply:
Thanks for the link. Now that I think about it, we are planning to remove our dependency on kubernetes code so probably best to not to user their code

Reply:
If the original intent was to match k8s label rules, it may be best to reuse the k8s label validation logic as well and not handcraft something similar. 
k8s.io/apimachinery/pkg/apis/meta/v1/validation has a `ValidateLabels()` function... 
Removing k8s as a dependency for deployment or code base? I would think that code reuse may still make sense, but perhaps I misunderstood the dependency removal goal.

Reply:
> If the original intent was to match k8s label rules, it may be best to reuse the k8s label validation logic as well and not handcraft something similar.

yeah, I'd agree with that.

> Removing k8s as a dependency for deployment or code base? I would think that code reuse may still make sense, but perhaps I misunderstood the dependency removal goal.

is using Istio outside of Kubernetes still a design goal?